# Fix Bazel dependencies.
#
# ```
# buildozer 'replace deps @eigen_archive//:eigen3 @eigen//:eigen' //third_party/tsl/...:*
# buildozer 'substitute deps @com_google_absl//(.*) @abseil-cpp//${1}' //third_party/tsl/...:*
# buildozer 'substitute deps @com_googlesource_code_re2//(.*) @re2//${1}' //third_party/tsl/...:*
# ```

diff --git third_party/ducc/ducc.BUILD third_party/ducc/ducc.BUILD
index 33b411e136..82dfb41482 100644
--- third_party/ducc/ducc.BUILD
+++ third_party/ducc/ducc.BUILD
@@ -64,7 +64,7 @@ cc_library(
     visibility = ["//visibility:private"],
     deps = [
         # Required for custom threadpool usage:
-        "@eigen_archive//:eigen3",
+        "@@eigen//:eigen",
         "@tsl//tsl/platform:mutex",
     ],
 )
@@ -80,6 +80,6 @@ cc_library(
     visibility = ["//visibility:public"],
     deps = [
         ":fft",
-        "@eigen_archive//:eigen3",
+        "@@eigen//:eigen",
     ],
 )
diff --git third_party/py/ml_dtypes/ml_dtypes.BUILD third_party/py/ml_dtypes/ml_dtypes.BUILD
index 6e3b504c62..12ec1c8886 100644
--- third_party/py/ml_dtypes/ml_dtypes.BUILD
+++ third_party/py/ml_dtypes/ml_dtypes.BUILD
@@ -10,7 +10,7 @@ package(
 cc_library(
     name = "float8",
     hdrs = ["include/float8.h"],
-    deps = ["@eigen_archive//:eigen3"],
+    deps = ["@@eigen//:eigen"],
 )
 
 cc_library(
@@ -23,7 +23,7 @@ cc_library(
     hdrs = ["include/mxfloat.h"],
     deps = [
         ":float8",
-        "@eigen_archive//:eigen3",
+        "@@eigen//:eigen",
     ],
 )
 
@@ -43,7 +43,7 @@ pybind_extension(
         ":float8",
         ":intn",
         ":mxfloat",
-        "@eigen_archive//:eigen3",
+        "@@eigen//:eigen",
         "@xla//third_party/py/numpy:headers",
     ],
 )
diff --git third_party/pybind11.BUILD third_party/pybind11.BUILD
index 7d284cc50f..25d57192e4 100644
--- third_party/pybind11.BUILD
+++ third_party/pybind11.BUILD
@@ -45,7 +45,7 @@ cc_library(
     includes = ["include"],
     strip_include_prefix = "include",
     deps = [
-        "@eigen_archive//:eigen3",
+        "@@eigen//:eigen",
         "@xla//third_party/python_runtime:headers",
     ],
 )
diff --git xla/BUILD xla/BUILD
index 925b050b43..8ad2fbb871 100644
--- xla/BUILD
+++ xla/BUILD
@@ -86,7 +86,7 @@ cc_library(
     deps = [
         ":types",
         "@com_google_absl//absl/base",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:bfloat16",
     ],
 )
@@ -99,7 +99,7 @@ xla_cc_test(
         "//xla/hlo/testlib:test",
         "//xla/tsl/platform:test_main",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:bfloat16",
     ],
 )
@@ -212,7 +212,7 @@ cc_library(
     visibility = internal_visibility([":friends"]),
     deps = [
         "@com_google_absl//absl/strings",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
     ],
 )
@@ -337,7 +337,7 @@ cc_library(
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:bfloat16",
         "@tsl//tsl/platform:casts",
         "@tsl//tsl/platform:ml_dtypes",
@@ -642,7 +642,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
         "@tsl//tsl/platform:platform_port",
     ],
@@ -783,7 +783,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -828,7 +828,7 @@ xla_cc_test(
         "@com_google_absl//absl/log:check",
         "@com_google_absl//absl/strings:string_view",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -853,7 +853,7 @@ xla_cc_test(
         "//xla/hlo/testlib:test",
         "//xla/tsl/platform:test_main",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
     ],
 )
@@ -906,7 +906,7 @@ xla_cc_test(
         "@com_google_absl//absl/log",
         "@com_google_absl//absl/types:span",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/backends/cpu/BUILD xla/backends/cpu/BUILD
index 67ebfd4f11..dffc07bbbb 100644
--- xla/backends/cpu/BUILD
+++ xla/backends/cpu/BUILD
@@ -30,7 +30,7 @@ filegroup(
 cc_library(
     name = "alignment",
     hdrs = ["alignment.h"],
-    deps = ["@eigen_archive//:eigen3"],
+    deps = ["@eigen"],
 )
 
 onednn_graph_cc_library(
diff --git xla/backends/cpu/nanort/BUILD xla/backends/cpu/nanort/BUILD
index ea5eb96980..ee1c7e945a 100644
--- xla/backends/cpu/nanort/BUILD
+++ xla/backends/cpu/nanort/BUILD
@@ -76,7 +76,7 @@ xla_cc_test(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings:string_view",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -118,7 +118,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:casts",
         "@tsl//tsl/platform:platform_port",
         "@tsl//tsl/profiler/lib:traceme",
diff --git xla/backends/cpu/runtime/BUILD xla/backends/cpu/runtime/BUILD
index 9a872677d3..c103c74382 100644
--- xla/backends/cpu/runtime/BUILD
+++ xla/backends/cpu/runtime/BUILD
@@ -74,7 +74,7 @@ cc_library(
     hdrs = ["thread_pool_task_runner.h"],
     deps = [
         ":thunk",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -102,7 +102,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -122,7 +122,7 @@ xla_cc_test(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:platform_port",
     ],
 )
@@ -154,7 +154,7 @@ cc_library(
         "@com_google_absl//absl/log:check",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -171,7 +171,7 @@ xla_cc_test(
         "@com_google_absl//absl/algorithm:container",
         "@com_google_absl//absl/cleanup",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -310,7 +310,7 @@ xla_cc_test(
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -425,7 +425,7 @@ cc_library(
         "//xla/tsl/framework/contraction:eigen_contraction_kernel",
         "//xla/tsl/framework/convolution:eigen_helpers",
         "//xla/tsl/platform:logging",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -450,7 +450,7 @@ cc_library(
         "@com_google_absl//absl/memory",
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings:str_format",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -484,7 +484,7 @@ xla_cc_test(
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/types:span",
         "@com_google_googletest//:gtest_main",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -660,7 +660,7 @@ cc_library(
         "@com_google_absl//absl/memory",
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings:str_format",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -716,7 +716,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/strings:string_view",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@llvm-project//mlir:AsmParser",
         "@llvm-project//mlir:IR",
         "@llvm-project//mlir:Support",
@@ -774,7 +774,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -794,7 +794,7 @@ xla_cc_test(
         "//xla/tsl/platform:statusor",
         "//xla/tsl/platform:test",
         "@com_google_absl//absl/strings",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:test_main",
     ],
 )
@@ -954,7 +954,7 @@ cc_library(
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/strings:string_view",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1130,7 +1130,7 @@ xla_cc_test(
         "//xla/tsl/platform:statusor",
         "//xla/tsl/platform:test",
         "@com_google_googletest//:gtest_main",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1154,7 +1154,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/types:span",
         "@ducc//:fft_wrapper",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1245,7 +1245,7 @@ cc_library(
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/container:fixed_array",
         "@com_google_absl//absl/status",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1261,7 +1261,7 @@ xla_cc_test(
         "//xla/tsl/platform:test_main",
         "@com_google_absl//absl/algorithm:container",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/backends/cpu/runtime/onednn/BUILD xla/backends/cpu/runtime/onednn/BUILD
index ec332b705a..615f75968e 100644
--- xla/backends/cpu/runtime/onednn/BUILD
+++ xla/backends/cpu/runtime/onednn/BUILD
@@ -69,7 +69,7 @@ onednn_graph_cc_library(
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/time",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@onednn//:mkl_dnn",
         "@tsl//tsl/profiler/lib:traceme",
     ],
@@ -81,7 +81,7 @@ onednn_graph_cc_library(
     deps = [
         ":onednn_interop",
         "//xla/backends/cpu/runtime:work_queue",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@onednn//:mkl_dnn",
     ],
 )
@@ -104,7 +104,7 @@ onednn_graph_cc_test(
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@onednn//:mkl_dnn",
     ],
 )
diff --git xla/backends/cpu/runtime/xnnpack/BUILD xla/backends/cpu/runtime/xnnpack/BUILD
index fc97639a34..c90f05e237 100644
--- xla/backends/cpu/runtime/xnnpack/BUILD
+++ xla/backends/cpu/runtime/xnnpack/BUILD
@@ -49,7 +49,7 @@ cc_library(
         "@com_google_absl//absl/log",
         "@com_google_absl//absl/log:check",
         "@com_google_absl//absl/strings:str_format",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@pthreadpool",
         "@tsl//tsl/platform:platform_port",
     ],
@@ -68,7 +68,7 @@ xla_cc_test(
         "//xla/tsl/platform:test_main",
         "@XNNPACK",
         "@com_google_absl//absl/algorithm:container",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@pthreadpool",
     ],
 )
@@ -127,7 +127,7 @@ xla_cc_test(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
         "@com_google_googletest//:gtest_main",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -176,7 +176,7 @@ xla_cc_test(
         "//xla/tsl/platform:test",
         "@com_google_absl//absl/strings",
         "@com_google_googletest//:gtest_main",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:platform_port",
     ],
 )
@@ -239,6 +239,6 @@ xla_cc_test(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/types:span",
         "@com_google_googletest//:gtest_main",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
diff --git xla/backends/gpu/codegen/triton/BUILD xla/backends/gpu/codegen/triton/BUILD
index b580664000..732f7ed4d4 100644
--- xla/backends/gpu/codegen/triton/BUILD
+++ xla/backends/gpu/codegen/triton/BUILD
@@ -758,7 +758,7 @@ xla_test(
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/strings",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@llvm-project//llvm:ir_headers",
         "@llvm-project//mlir:IR",
         "@llvm-project//mlir:Pass",
diff --git xla/backends/gpu/runtime/BUILD xla/backends/gpu/runtime/BUILD
index fcc8b126d4..4a90b04c83 100644
--- xla/backends/gpu/runtime/BUILD
+++ xla/backends/gpu/runtime/BUILD
@@ -1657,7 +1657,7 @@ cc_library(
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/ffi/BUILD xla/ffi/BUILD
index 0964189695..027daa8bb8 100644
--- xla/ffi/BUILD
+++ xla/ffi/BUILD
@@ -182,7 +182,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/strings:string_view",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -230,7 +230,7 @@ xla_cc_test(
         "@com_google_absl//absl/strings:string_view",
         "@com_google_absl//absl/types:span",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:status_matchers",
         "@tsl//tsl/platform:statusor",
diff --git xla/ffi/api/BUILD xla/ffi/api/BUILD
index f42c71a4e2..0ed5bf54a8 100644
--- xla/ffi/api/BUILD
+++ xla/ffi/api/BUILD
@@ -105,6 +105,6 @@ xla_cc_test(
         "@com_google_absl//absl/strings:string_view",
         "@com_google_absl//absl/synchronization",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
diff --git xla/hlo/evaluator/BUILD xla/hlo/evaluator/BUILD
index d3992af818..e4c3789750 100644
--- xla/hlo/evaluator/BUILD
+++ xla/hlo/evaluator/BUILD
@@ -89,7 +89,7 @@ cc_library(
         "@com_google_absl//absl/strings:cord",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
         "@tsl//tsl/platform:platform_port",
     ],
diff --git xla/hlo/parser/BUILD xla/hlo/parser/BUILD
index 67e1ba79ad..0571bd1262 100644
--- xla/hlo/parser/BUILD
+++ xla/hlo/parser/BUILD
@@ -59,7 +59,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:protobuf",
     ],
 )
diff --git xla/pjrt/BUILD xla/pjrt/BUILD
index af2a8dc9fe..b780a81c39 100644
--- xla/pjrt/BUILD
+++ xla/pjrt/BUILD
@@ -914,7 +914,7 @@ xla_cc_test(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:statusor",
         "@tsl//tsl/platform:test_benchmark",
diff --git xla/pjrt/cpu/BUILD xla/pjrt/cpu/BUILD
index ed72fa942e..2040f6f71b 100644
--- xla/pjrt/cpu/BUILD
+++ xla/pjrt/cpu/BUILD
@@ -222,7 +222,7 @@ cc_library(
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",  # TODO(zhangqiaorjc): Remove if use TFRT threadpool.
+        "@eigen",  # TODO(zhangqiaorjc): Remove if use TFRT threadpool.
         "@llvm-project//mlir:IR",
         "@tsl//tsl/platform:casts",
         "@tsl//tsl/platform:denormal",
diff --git xla/pjrt/gpu/tfrt/BUILD xla/pjrt/gpu/tfrt/BUILD
index a27b6d4b66..c25d6c8118 100644
--- xla/pjrt/gpu/tfrt/BUILD
+++ xla/pjrt/gpu/tfrt/BUILD
@@ -115,7 +115,7 @@ cc_library(
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@llvm-project//mlir:IR",
         "@tsl//tsl/platform:casts",
         "@tsl//tsl/platform:fingerprint",
diff --git xla/service/BUILD xla/service/BUILD
index 1e6dd7ff39..9965962740 100644
--- xla/service/BUILD
+++ xla/service/BUILD
@@ -1052,7 +1052,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:platform_port",
         "@tsl//tsl/platform:statusor",
@@ -4404,7 +4404,7 @@ cc_library(
         "@com_google_absl//absl/strings:string_view",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/service/cpu/BUILD xla/service/cpu/BUILD
index 144fd426e3..12c41e933b 100644
--- xla/service/cpu/BUILD
+++ xla/service/cpu/BUILD
@@ -703,7 +703,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:denormal",
         "@tsl//tsl/platform:setround",
     ],
@@ -1127,7 +1127,7 @@ cc_library(
         "//xla/backends/cpu/runtime:convolution_thunk_internal",
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/synchronization",  # build_cleaner: keep
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1143,7 +1143,7 @@ cc_library(
         "//xla/backends/cpu/runtime:convolution_thunk_internal",
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/synchronization",  # build_cleaner: keep
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1180,7 +1180,7 @@ cc_library(
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/base:dynamic_annotations",
         "@com_google_absl//absl/synchronization",  # build_cleaner: keep
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1196,7 +1196,7 @@ cc_library(
         "//xla:executable_run_options",
         "//xla/tsl/platform:dynamic_annotations",
         "@com_google_absl//absl/base",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:logging",
         "@tsl//tsl/platform:types",
     ] + acl_deps(),
@@ -1218,7 +1218,7 @@ cc_library(
         "//xla/tsl/framework/convolution:eigen_helpers",
         "//xla/tsl/platform:dynamic_annotations",
         "@com_google_absl//absl/base",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:logging",
         "@tsl//tsl/platform:types",
     ] + acl_deps(),
@@ -1234,7 +1234,7 @@ cc_library(
         "//xla/backends/cpu/runtime:convolution_thunk_internal",
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/synchronization",  # build_cleaner: keep
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1248,7 +1248,7 @@ cc_library(
         "//xla/backends/cpu/runtime:convolution_thunk_internal",
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/synchronization",  # build_cleaner: keep
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1273,7 +1273,7 @@ cc_library(
     deps = [
         "//xla/tsl/framework/contraction:eigen_contraction_kernel_no_mkl",
         "@com_google_absl//absl/base:core_headers",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
     ],
 )
@@ -1286,7 +1286,7 @@ cc_library(
     visibility = ["//visibility:public"],
     deps = [
         ":runtime_single_threaded_matmul_impl",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
     ],
 )
@@ -1301,7 +1301,7 @@ cc_library(
         ":runtime_single_threaded_matmul_impl",
         "//xla/tsl/framework/contraction:eigen_contraction_kernel_no_mkl",
         "@com_google_absl//absl/base:core_headers",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
     ],
 )
@@ -1315,7 +1315,7 @@ cc_library(
     deps = [
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/base:dynamic_annotations",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1350,7 +1350,7 @@ xla_cc_test(
         "//xla/service:custom_call_status_internal",
         "//xla/tests:xla_internal_test_main",
         "@com_google_absl//absl/strings:str_format",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:logging",
         "@tsl//tsl/platform:test",
@@ -1705,7 +1705,7 @@ cc_library(
         "//xla/hlo/ir:hlo",
         "//xla/tsl/platform:env",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:platform_port",
     ] + mkl_deps(),
@@ -1759,7 +1759,7 @@ cc_library(
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/base:dynamic_annotations",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:logging",
         "@tsl//tsl/platform:platform_port",
@@ -1784,7 +1784,7 @@ cc_library(
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/base:dynamic_annotations",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:logging",
         "@tsl//tsl/platform:platform_port",
@@ -1810,7 +1810,7 @@ cc_library(
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/base:dynamic_annotations",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:platform_port",
     ] + mkl_deps(),
@@ -1835,7 +1835,7 @@ cc_library(
         "@com_google_absl//absl/base:core_headers",
         "@com_google_absl//absl/base:dynamic_annotations",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:platform_port",
     ] + mkl_deps(),
@@ -1884,7 +1884,7 @@ cc_library(
         "@com_google_absl//absl/algorithm:container",
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/synchronization",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:env",
         "@tsl//tsl/platform:logging",
         "@tsl//tsl/platform:platform_port",
diff --git xla/service/gpu/BUILD xla/service/gpu/BUILD
index 7258b3aae0..803af5469e 100644
--- xla/service/gpu/BUILD
+++ xla/service/gpu/BUILD
@@ -1074,7 +1074,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
         "@tsl//tsl/platform:statusor",
     ],
@@ -2401,7 +2401,7 @@ cc_library(
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
         "@tsl//tsl/platform:status",
         "@tsl//tsl/platform:statusor",
diff --git xla/service/gpu/tests/BUILD xla/service/gpu/tests/BUILD
index 4f6b4d3cc3..d0ecea3304 100644
--- xla/service/gpu/tests/BUILD
+++ xla/service/gpu/tests/BUILD
@@ -572,7 +572,7 @@ xla_test(
         "@com_google_absl//absl/log:check",
         "@com_google_absl//absl/strings",
         "@com_google_googletest//:gtest_main",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/stream_executor/BUILD xla/stream_executor/BUILD
index 9c680f5a56..84f1dc8352 100644
--- xla/stream_executor/BUILD
+++ xla/stream_executor/BUILD
@@ -378,7 +378,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",  # buildcleaner: keep
+        "@eigen",  # buildcleaner: keep
         "@tsl//tsl/platform:errors",
         "@tsl//tsl/platform:logging",
         "@tsl//tsl/platform:ml_dtypes",
diff --git xla/stream_executor/cuda/BUILD xla/stream_executor/cuda/BUILD
index 421a4768ff..bf14a91a13 100644
--- xla/stream_executor/cuda/BUILD
+++ xla/stream_executor/cuda/BUILD
@@ -305,7 +305,7 @@ cc_library(
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@local_config_cuda//cuda:cuda_headers",
         "@tsl//tsl/platform:errors",
         "@tsl//tsl/platform:logging",
@@ -472,7 +472,7 @@ cc_library(
         "@com_google_absl//absl/time",
         "@com_google_absl//absl/types:span",
         "@cudnn_frontend_archive//:cudnn_frontend",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@local_config_cuda//cuda:cuda_headers",
         "@local_config_cuda//cuda:cudnn_header",  # build_cleaner: keep
         "@tsl//tsl/platform:tensor_float_32_hdr_lib",
diff --git xla/stream_executor/rocm/BUILD xla/stream_executor/rocm/BUILD
index 4581e3223e..97f7a1d52c 100644
--- xla/stream_executor/rocm/BUILD
+++ xla/stream_executor/rocm/BUILD
@@ -399,7 +399,7 @@ cc_library(
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@local_config_rocm//rocm:rocm_headers",
     ],
     alwayslink = True,
@@ -541,7 +541,7 @@ cc_library(
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@local_config_rocm//rocm:rocm_headers",
         "@tsl//tsl/platform:dso_loader",
         "@tsl//tsl/platform:hash",
@@ -703,7 +703,7 @@ cc_library(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/time",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@local_config_rocm//rocm:rocm_headers",
         "@tsl//tsl/platform:dso_loader",
         "@tsl//tsl/platform:ml_dtypes",
@@ -1159,7 +1159,7 @@ rocm_library(
         "//xla/stream_executor/rocm:rocm_status",
         "@com_google_absl//absl/base",
         "@com_google_absl//absl/status",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@local_config_rocm//rocm:rocprim",
         "@tsl//tsl/platform:bfloat16",
     ],
diff --git xla/tests/BUILD xla/tests/BUILD
index dc0b577a3d..912ffd7b46 100644
--- xla/tests/BUILD
+++ xla/tests/BUILD
@@ -457,7 +457,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1044,7 +1044,7 @@ xla_test(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:ml_dtypes",
         "@tsl//tsl/platform:statusor",
         "@tsl//tsl/platform:test",
@@ -1358,7 +1358,7 @@ xla_test(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1394,7 +1394,7 @@ xla_test(
         "//xla/hlo/builder:padding",
         "//xla/hlo/builder:xla_builder",
         "//xla/tsl/platform:test",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1426,7 +1426,7 @@ xla_test(
         "//xla/hlo/builder:padding",
         "//xla/hlo/builder:xla_builder",
         "//xla/tsl/platform:test",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1467,7 +1467,7 @@ xla_test(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1497,7 +1497,7 @@ xla_test(
         "//xla/hlo/builder:padding",
         "//xla/hlo/builder:xla_builder",
         "//xla/tsl/platform:test",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1531,7 +1531,7 @@ xla_test(
         "@com_google_absl//absl/status:statusor",
         "@com_google_absl//absl/strings",
         "@com_google_googletest//:gtest",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1560,7 +1560,7 @@ xla_test(
         "//xla/hlo/builder:padding",
         "//xla/hlo/builder:xla_builder",
         "//xla/tsl/platform:test",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -1746,7 +1746,7 @@ xla_test(
         "//xla/tsl/platform:test",
         "@com_google_absl//absl/log:check",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -2229,7 +2229,7 @@ xla_test(
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
@@ -2386,7 +2386,7 @@ xla_test(
         "//xla/hlo/testlib:test",
         "//xla/tests:xla_test_backend_predicates",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:protobuf",
         "@tsl//tsl/platform:test",
     ],
@@ -3070,7 +3070,7 @@ xla_test(
         "@com_google_absl//absl/log",
         "@com_google_absl//absl/types:span",
         "@com_google_benchmark//:benchmark",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform:protobuf",
     ],
 )
diff --git xla/tsl/framework/BUILD xla/tsl/framework/BUILD
index 4039092924..fd9b9f388f 100644
--- xla/tsl/framework/BUILD
+++ xla/tsl/framework/BUILD
@@ -327,7 +327,7 @@ cc_library(
     compatible_with = get_compatible_with_portable(),
     visibility = ["//visibility:public"],
     deps = [
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/tsl/framework/contraction/BUILD xla/tsl/framework/contraction/BUILD
index f34116d0b6..5691e0e0c4 100644
--- xla/tsl/framework/contraction/BUILD
+++ xla/tsl/framework/contraction/BUILD
@@ -107,7 +107,7 @@ cc_library(
         "//xla/tsl/framework/fixedpoint",
         "//xla/tsl/platform:dynamic_annotations",
         "@com_google_absl//absl/base",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ] + select({
         ":disable_onednn_contraction_kernel_config": [],
         "//xla/tsl:android_x86": [],
@@ -144,7 +144,7 @@ cc_library(
         "//xla/tsl/framework/fixedpoint",
         "//xla/tsl/platform:dynamic_annotations",
         "@com_google_absl//absl/base",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/tsl/framework/convolution/BUILD xla/tsl/framework/convolution/BUILD
index 9687b8a9fd..5fcd58a2f4 100644
--- xla/tsl/framework/convolution/BUILD
+++ xla/tsl/framework/convolution/BUILD
@@ -48,7 +48,7 @@ cc_library(
         ":eigen_convolution_helpers",
         ":eigen_spatial_convolutions-inl",
         "//xla/tsl/framework/contraction:eigen_contraction_kernel",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/tsl/framework/fixedpoint/BUILD xla/tsl/framework/fixedpoint/BUILD
index c1ec1d9e17..da4323c05e 100644
--- xla/tsl/framework/fixedpoint/BUILD
+++ xla/tsl/framework/fixedpoint/BUILD
@@ -29,7 +29,7 @@ cc_library(
     compatible_with = get_compatible_with_portable(),
     deps = [
         "//xla/tsl/framework:fixedpoint_types",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/tsl/lib/random/BUILD xla/tsl/lib/random/BUILD
index 3b4b420f15..9a4cda459a 100644
--- xla/tsl/lib/random/BUILD
+++ xla/tsl/lib/random/BUILD
@@ -44,7 +44,7 @@ cc_library(
         "//xla/tsl/platform:macros",
         "//xla/tsl/platform:types",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
     alwayslink = 1,
 )
diff --git xla/tsl/platform/BUILD xla/tsl/platform/BUILD
index a78c35880b..ade5ae0c90 100644
--- xla/tsl/platform/BUILD
+++ xla/tsl/platform/BUILD
@@ -699,7 +699,7 @@ cc_library(
     compatible_with = get_compatible_with_portable(),
     deps = [
         ":types",
-        "@eigen_archive//:eigen3",
+        "@eigen",
     ],
 )
 
diff --git xla/tsl/platform/default/BUILD xla/tsl/platform/default/BUILD
index fed3f1cdc5..691e775d85 100644
--- xla/tsl/platform/default/BUILD
+++ xla/tsl/platform/default/BUILD
@@ -162,7 +162,7 @@ cc_library(
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/synchronization",
         "@com_google_absl//absl/types:span",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform",
         "@tsl//tsl/platform:blocking_counter",
         "@tsl//tsl/platform:context",
diff --git xla/tsl/platform/windows/BUILD xla/tsl/platform/windows/BUILD
index e8c1cb4f44..61d4654522 100644
--- xla/tsl/platform/windows/BUILD
+++ xla/tsl/platform/windows/BUILD
@@ -57,7 +57,7 @@ cc_library(
         "//xla/tsl/protobuf:error_codes_proto_impl_cc",
         "@com_google_absl//absl/functional:any_invocable",
         "@com_google_absl//absl/time",
-        "@eigen_archive//:eigen3",
+        "@eigen",
         "@tsl//tsl/platform",
         "@tsl//tsl/platform:blocking_counter",
         "@tsl//tsl/platform:context",
